app/services/group_service.rb

```diff
@@ -41,6 +41,7 @@ def self.update(group:, params:, actor:)
 
   def self.archive(group:, actor:)
     actor.ability.authorize! :archive, group
+    SubscriptionService.new(group, actor).end_subscription!
```

Comment:
It was initially, but it meant the event was being fired after the record was deleted and we need to reference the record to get the chargify_subscription_id. We need to call end_subscription! before the group is archived.